tpm2/credential_activation.go

```diff
@@ -0,0 +1,134 @@
+package tpm2
+
+import (
+	"crypto"
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/hmac"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/sha256"
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/google/go-tpm/tpmutil"
+)
+
+// GenerateCredentialActivation creates a wrapped secret for use in credential activation.
+// This function implements the minimum necessary to compute activations for RSA EKs
+// compliant with TCG 2.0 EK Credential Profile specification, revision 14.
+// This should be all windows8+ cerified hardware implementing a TPM 2.0.
+// Specifically, the constraints are:
```

Comment:
I definitely have to restrict key type as to support ECC keys is a completely different KDF function and a bunch of other mess (ECC keys cant just use a random seed).

I disagree this is horribly specialized; this should work for the vast majority of EK deployments out in the world (those that have ECC EKs also have RSA EKs alongside).

I'll look into supporting other hash algs and whatnot, but given it took nearly 3 days to get this working, I cant make any promises.

Reply:
OK, i understand ECC keys would be too much work.
But please remove the other 2 limitations (hash alg and block size). And make a distinction between ECC key (should return "key type not supported yet") and anything else (should return "key type invalid")

Reply:
Will do. Note that i will have no way to test this, I am yet to encounter an EK in the wild that has any other properties (presumably because everyone is trying to support windows).

Reply:
Ive removed the hash alg and block size constraints.